src/internal/connector/graph_connector.go

```diff
@@ -232,19 +233,55 @@ func identifySite(item any) (string, string, error) {
 		return "", "", errKnownSkippableCase
 	}
 
-	return *m.GetName(), *m.GetId(), nil
+	return *m.GetWebUrl(), *m.GetId(), nil
 }
 
-// GetSites returns the siteIDs of sharepoint sites within tenant.
-func (gc *GraphConnector) GetSites() []string {
+// GetSiteWebURLs returns the WebURLs of sharepoint sites within the tenant.
+func (gc *GraphConnector) GetSiteWebURLs() []string {
 	return buildFromMap(true, gc.Sites)
 }
 
-// GetSiteIds returns the M365 id for the user
-func (gc *GraphConnector) GetSiteIds() []string {
+// GetSiteIds returns the canonical site IDs in the tenant
+func (gc *GraphConnector) GetSiteIDs() []string {
 	return buildFromMap(false, gc.Sites)
 }
 
+// UnionSiteIDsAndWebURLs reduces the id and url slices into a single slice of site IDs.
+// WebURLs will run as a path-suffix style matcher.  Callers may provide partial urls, though
+// each element in the url must fully match.  Ex: the webURL value "foo" will match "www.ex.com/foo",
+// but not match "www.ex.com/foobar".
+// The returned IDs are reduced to a set of unique values.
```

Comment:
What is the unique value returned? Comment could provide an example or be made more succinct.